packages/flutter/lib/src/gestures/drag_details.dart

```diff
@@ -182,6 +185,16 @@ class DragUpdateDetails {
   /// Defaults to [globalPosition] if not specified in the constructor.
   final Offset localPosition;
 
+
+  /// Indicates whether the drag gesture is within the boundary.
+  ///
+  /// This property is true if the drag gesture is within the boundary or has no bounds;
+  /// otherwise, false.It can be used to determine whether to take certain actions
+  /// based on whetherthe drag gesture is within the boundary.
+  ///
+  /// Defaults to true if not specified in the constructor.
```

Comment:
It sounds like this should default to `null` if no drag boundary is defined. If not then users might incorrectly think that their drag is within bounds when it is not.

Reply:
Hello, I have re-examined the code here and found that it's not really necessary to call `outOfBoundary`, so I removed it. I put the boundary information into `DragStartDetails` and `DragUpdateDetails`, so there's no need to add some bad APIs, making it a bit simpler. Please review it again, thank you.